R/rmarkdown.R

```diff
@@ -20,13 +20,23 @@ render_rmarkdown <- function(pkg, input, output, ..., copy_images = TRUE, quiet
     intermediates_dir = tempdir(),
     encoding = "UTF-8",
     envir = globalenv(),
+    seed = seed,
     ...,
     quiet = quiet
   )
 
   path <- tryCatch(
     callr::r_safe(
-      function(...) rmarkdown::render(...),
+      function(seed, envir, ...) {
+        if (!is.null(seed)) {
+          set.seed(seed)
+          envir$.Random.seed <- .GlobalEnv$.Random.seed
```

Comment:
What does this line do?

Reply:
I don't remember exactly why I added both lines, `set.seed(seed)` and `envir$.Random.seed <- .GlobalEnv$.Random.seed`... I think, `set.seed()` alters the global env; but since the current env was created before the seed in the global env was defined, we have to manually overwrite `.Random.seed` in the current env with the newly set seed.

IIRC, I first only added `set.seed(seed)`, which had no effect by itself. Then I added `envir$.Random.seed <- .GlobalEnv$.Random.seed` and it worked as supposed.

Reply:
Oooh, maybe it's because `envir` is copied from the parent session, and hence gets the `.Random.seed` from the wrong global environment.

Reply:
Ok, will do later (gotta go now).

Reply:
> Could you please add a test along the same lines as in https://github.com/r-lib/pkgdown/pull/2355
> Can you please add a comment explaining the seed assignment too?

Both done :)